src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_validators.py

```diff
@@ -284,6 +284,14 @@ def validate_client_parameters(cmd, namespace):
     if not namespace.account_endpoint:
         namespace.account_endpoint = cmd.cli_ctx.config.get('batch', 'endpoint', None)
 
+    # Verify all values are populated and display readable error
+    if not namespace.account_name:
+        raise ValueError("Account name not set.")
+    if not namespace.account_key:
+        raise ValueError("Account key not set.")
+    if not namespace.account_endpoint:
+        raise ValueError("Account endpoint not set.")
```

Comment:
Same thing here. As a user, this would be VERY annoying. Are these three always required? If so, why are they not just required parameters?  Then the CLI would automatically handle it?

Reply:
They are only required if the user doesn't 'login' to their account, or have them set as environment variables. In those cases they are optional. We make use of the CLI's config to do this.  Does the core CLI have a way to say required if not in config, if so we could do that.

Reply:
@bgklein not really. If something is marked as required but is using configured_default to allow the config or environment variable, then, if either of those are set, the CLI should no longer display [Required] and should show the default. However, since it could also come from being logged in, there's no infrastructure-supported way to manage that. The above Python snippet is generally what we do in these situations.